### PR TITLE
[parachain-tracer] Introduce top level restart

### PR DESCRIPTION
**Pull request by AndreiEres**
Fixes https://github.com/paritytech/polkadot-introspector/issues/769

Introduces two types of shutdown: 
1. Graceful – when we stop the application with ctrl-c.
2. Restart – something bad happened, generaly lost the connection to RPC node.

On restart we reinitialize the application from the beginning. Same rules for retry as in other parts of application applied. 

Additionally in the other PR was updated subxt, which added own reconnection to RPC-nodes.

### CODE REVIEW

**Comment by s0me0ne-unkn0wn** [comment]
That sounds like healing symptoms instead of healing the disease, doesn't it? Is there any objective reason why the root cause cannot be fixed?

**Comment by sandreim** [comment]
> That sounds like healing symptoms instead of healing the disease, doesn't it? Is there any objective reason why the root cause cannot be fixed?

Yes indeed. Do we know why it happens? Maybe something else needs a fix.

**Comment by AndreiEres** [comment]
RPC nodeы drop connections after a while or can be unreachable. But I agree, that it's too straightforward.

**Review by alexggh — APPROVED** [review]
Looks like an improvement to the current state.

However, I do second the previous comments that restarting on errors might mask other bugs.

File: parachain-tracer/src/main.rs [hunk 1]
```diff
@@ -312,9 +317,10 @@ impl ParachainTracer {
 									trackers.len(), futures.len());
 								match reason {
 									TerminationReason::Normal => break,
-									TerminationReason::Abnormal(code, info) => {
+									TerminationReason::Abnormal( info) => {
```
Comment (s0me0ne-unkn0wn) [inline parachain-tracer/src/main.rs:320]:
```suggestion
									TerminationReason::Abnormal(info) => {
```

**Review by s0me0ne-unkn0wn — APPROVED** [review]
Looks good as a solution for the current problem :+1: RPC is evil anyway, let's just hope we'll get rid of it altogether at some point.

**Review by sandreim — APPROVED** [review]

**Pull request merged**